### ISSUE TITLE
Rewrite ORSSerialPort itself in Swift

### ISSUE THREAD

**Issue by armadsen** [armadsen/ORSSerialPort#92]
This is a major change for 3.0. My intention is to maintain Objective-C compatibility and keep breaking API changes to a minimum.

**Comment by armadsen** [comment]
Don't close this without testing sleep/wake and close-on-quit functionality in the new Swift `SerialPortManager`.

**Comment by kouky** [comment]
Wow, this would be amazing.

**Comment by armadsen** [comment]
:+1: 

I'm well underway. The code in the 3.0 branch includes all classes except `ORSSerialPort` itself rewritten in Swift. I'm in the process of rewriting `ORSSerialPort` (aka. `SerialPort`) itself now.

Early feedback would be most welcome.

**Comment by kouky** [comment]
@armadsen yes I can see you've been busy, wish I saw this earlier so I could have helped in the conversion.

**Comment by marciok** [comment]
+1 
Looking forward, for the rewrite.

**Comment by ladislas** [comment]
how can we help?
